File: sqlbuilder/sqlbuilder/UpdateAssignment.java
package com.clickability.cms.dataaccess.sqlbuilder;

import java.util.Date;

public class UpdateAssignment implements Expression {
	
	public final String operand;
	public final Binding value;
	public final String operator;
	public final String column;
	
	private UpdateAssignment(String column, String operand, String operator, Binding value) {
		this.operand = operand;
		this.value = value;
		this.operator = operator;
		this.column = column;
	}
	
	public UpdateAssignment(String column, Binding value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, Boolean value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Boolean value) {
		this(column, operand, operator, value == null ? new Binding(Boolean.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Byte value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Byte value) {
		this(column, operand, operator, value == null ? new Binding(Byte.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Character value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Character value) {
		this(column, operand, operator, value == null ? new Binding(Character.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Short value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Short value) {
		this(column, operand, operator, value == null ? new Binding(Short.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Integer value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Integer value) {
		this(column, operand, operator, value == null ? new Binding(Integer.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Long value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Long value) {
		this(column, operand, operator, value == null ? new Binding(Long.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Float value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Float value) {
		this(column, operand, operator, value == null ? new Binding(Float.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Double value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Double value) {
		this(column, operand, operator, value == null ? new Binding(Double.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, String value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, String value) {
		this(column, operand, operator, value == null ? new Binding(String.class) : new Binding(value));
	}
	
	public UpdateAssignment(String column, Date value) {
		this(column, null, null, value);
	}
	
	public UpdateAssignment(String column, String operand, String operator, Date value) {
		this(column, operand, operator, value == null ? new Binding(Date.class) : new Binding(value));
	}

	@Override
	public String get() {
		StringBuilder sb = new StringBuilder();
		sb.append(this.column);
		sb.append("=");
		if (this.operand != null) {
			sb.append(this.operand);
		}
		if (this.operator != null) {
			sb.append(this.operator);
		}
		sb.append("?");
		return sb.toString();
	}
}

File: java/queryBuilder/UpdateAssignment.kt
package com.clickability.cms.dataaccess.sqlbuilder

import java.util.Date

class UpdateAssignment private constructor(val column: String, val operand: String?, val operator: String?, val value: Binding) : Expression {

    constructor(column: String, value: Binding) : this(column, null, null, value) {}

    constructor(column: String, value: Boolean) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Boolean?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(Boolean::class.java)) {
    }

    constructor(column: String, value: Byte) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Byte?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(Byte::class.java)) {
    }

    constructor(column: String, value: Character) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Character?) : this(column, operand, operator, if (value == null) Binding(Character::class.java) else Binding(value)) {}

    constructor(column: String, value: Short) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Short?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(Short::class.java)) {
    }

    constructor(column: String, value: Integer) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Integer?) : this(column, operand, operator, if (value == null) Binding(Integer::class.java) else Binding(value)) {}

    constructor(column: String, value: Long) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Long?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(Long::class.java)) {
    }

    constructor(column: String, value: Float) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Float?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(Float::class.java)) {
    }

    constructor(column: String, value: Double) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Double?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(Double::class.java)) {
    }

    constructor(column: String, value: String) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: String?) : this(column, operand, operator, value?.let { Binding(it) }
            ?: Binding(String::class.java)) {
    }

    constructor(column: String, value: Date) : this(column, null, null, value) {}

    constructor(column: String, operand: String?, operator: String?, value: Date?) : this(column, operand, operator, if (value == null) Binding(Date::class.java) else Binding(value)) {}

    @Override
    fun get(): String {
        val sb = StringBuilder()
        sb.append(this.column)
        sb.append("=")
        if (this.operand != null) {
            sb.append(this.operand)
        }
        if (this.operator != null) {
            sb.append(this.operator)
        }
        sb.append("?")
        return sb.toString()
    }
}
